File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> group = new HashMap<>();

        for (String word: strs) {
            String sortedWord = word.chars()
                                        .sorted()
                                        .collect(StringBuilder::new, 
                                                    StringBuilder::appendCodePoint, 
                                                    StringBuilder::append)
                                        .toString();

            if (!group.containsKey(sortedWord)) {
                group.put(sortedWord, new ArrayList<>());
            }

            group.get(sortedWord).add(word);
        }

        return new ArrayList<>(group.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = {}

        for string in strs:
            sorted_string = "".join(sorted(string))

            if not anagrams.get(sorted_string):
                anagrams[sorted_string] = []

            anagrams[sorted_string].append(string)

        return list(anagrams.values())